evalml/tests/pipeline_tests/test_time_series_pipeline.py

```diff
@@ -1060,6 +1060,92 @@ def test_binary_predict_pipeline_objective_mismatch(
         binary_pipeline.predict(X[30:32], "precision micro", X[:30], y[:30])
 
 
+@pytest.mark.parametrize(
+    "problem_type",
+    [
+        ProblemTypes.TIME_SERIES_BINARY,
+        ProblemTypes.TIME_SERIES_MULTICLASS,
+        ProblemTypes.TIME_SERIES_REGRESSION,
+    ],
+)
+def test_time_series_pipeline_no_ytrain_predict_error(
+    problem_type,
+    ts_data_binary,
+    ts_data_multi,
+    ts_data,
+    time_series_binary_classification_pipeline_class,
+    time_series_multiclass_classification_pipeline_class,
+    time_series_regression_pipeline_class,
+):
+    if problem_type == ProblemTypes.TIME_SERIES_BINARY:
+        X, y = ts_data_binary
+        clf = time_series_binary_classification_pipeline_class(
+            parameters={
+                "Logistic Regression Classifier": {"n_jobs": 1},
+                "pipeline": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+                "Time Series Featurizer": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+            }
+        )
+
+    elif problem_type == ProblemTypes.TIME_SERIES_MULTICLASS:
+        X, y = ts_data_multi
+        clf = time_series_multiclass_classification_pipeline_class(
+            parameters={
+                "Logistic Regression Classifier": {"n_jobs": 1},
+                "pipeline": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+                "Time Series Featurizer": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+            }
+        )
+    else:
+        X, y = ts_data
+        clf = time_series_regression_pipeline_class(
+            parameters={
+                "Random Forest Regressor": {"n_jobs": 1},
+                "pipeline": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+                "Time Series Featurizer": {
+                    "gap": 0,
+                    "max_delay": 0,
+                    "time_index": "date",
+                    "forecast_horizon": 10,
+                },
+            }
+        )
+
+    X, y = pd.DataFrame(X), pd.Series(y)
+    X_train, y_train = X.iloc[:21], y.iloc[:21]
+    clf.fit(X_train, y_train)
+    with pytest.raises(
+        ValueError,
+        match="Make sure to have a value for both X_train and y_train when calling predict",
+    ):
+        clf.predict(X_train)
```

Comment:
It looks like if X_train is None, the `convert_to_woodwork()` function makes X_train a `pd.DataFrame()` and infers its type from there. It doesn't do the same for y_train which is why this initial error occurs. 

So probably, instead of checking both X_train and y_train, I just need to check y_train since if X_train is null the code still runs (though it does make X_train a blank dataframe).